Expensify/App
# Fix iPad overlay / click away handler issues

### Details

Simplifies some logic around the "click away handler" which is necessary for "wide screen" views to dismiss RH Modals.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2127

### Tests
### QA Steps
1. Open settings or profile pages
2. Verify the area to the left of the modal turns greyed out
3. Click/tap the area to the left of the modal and verify the modal is dismissed
4. Verify all links on the settings page can be accessed

### Tested On

- [x] Mobile Web (N/A)
- [x] iOS (iPad)
- [x] Android (Tablet)

### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
